### PR TITLE
fix(compiler): Correct the warning suggestion for Float32 And Float64 values

### PR DESCRIPTION
**Pull request by spotandjake**
This pr changes the error message in the case of a constant number being used with `Float32.fromNumber` or `Float64.fromNumber`.

Instead of `5f` or `5d` its now `5.f` and `5.d`

Closes: #1580

### CODE REVIEW

**Comment by alex-snezhko** [comment]
Would this work properly for something like `Float32.fromNumber(4.1)`? It seems this case would cause `4.1.f` to get suggested

**Comment by spotandjake** [comment]
> Would this work properly for something like `Float32.fromNumber(4.1)`? It seems this case would cause `4.1.f` to get suggested

This was fixed in #6d8ca74

**Comment by phated** [comment]
@spotandjake I'm not sure why the CI failed (it just said encountered a problem), but can you please add a regression test? It'll be a `assertCompilerError` in `compiler/test/suites/numbers.re`

File: compiler/test/suites/numbers.re [hunk 1]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
     "Overflow: Number overflow",
   );
+  // well-formedness errors
+  test("float32_fromNumber_err1", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF32("5"))).toMatch(
+      "5.f",
+    )
+  });
+  test("float32_fromNumber_err2", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF32("5.0"))).toMatch(
```
Comment (phated) [inline compiler/test/suites/numbers.re:146]:
```suggestion
    expect.string(Warnings.message(FromNumberLiteralF32("5."))).toMatch(
```

File: compiler/test/suites/numbers.re [hunk 2]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+      "5.5f",
+    )
+  });
+  test("float64_fromNumber_err1", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF64("5"))).toMatch(
+      "5.d",
+    )
+  });
+  test("float64_fromNumber_err2", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF64("5.0"))).toMatch(
```
Comment (phated) [inline compiler/test/suites/numbers.re:161]:
```suggestion
    expect.string(Warnings.message(FromNumberLiteralF64("5."))).toMatch(
```

File: compiler/test/suites/numbers.re [hunk 3]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
     {|include "uint64"; let n = 0x1ffffffffffffffff; print(Uint64.fromNumber(n))|},
     "Overflow: Number overflow",
   );
+  // well-formedness errors
```
Comment (phated) [inline compiler/test/suites/numbers.re:139]:
```suggestion
  // well-formedness warnings
```

File: compiler/test/suites/numbers.re [hunk 4]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+  test("float32_fromNumber_err1", ({expect}) => {
```
Comment (phated) [inline compiler/test/suites/numbers.re:140]:
```suggestion
  test("float32_fromNumber_warn1", ({expect}) => {
```

File: compiler/test/suites/numbers.re [hunk 5]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+    expect.string(Warnings.message(FromNumberLiteralF32("5"))).toMatch(
+      "5.f",
+    )
+  });
+  test("float32_fromNumber_err2", ({expect}) => {
```
Comment (phated) [inline compiler/test/suites/numbers.re:145]:
```suggestion
  test("float32_fromNumber_warn2", ({expect}) => {
```

File: compiler/test/suites/numbers.re [hunk 6]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+    expect.string(Warnings.message(FromNumberLiteralF32("5.0"))).toMatch(
+      "5.0f",
+    )
+  });
+  test("float32_fromNumber_err3", ({expect}) => {
```
Comment (phated) [inline compiler/test/suites/numbers.re:150]:
```suggestion
  test("float32_fromNumber_warn3", ({expect}) => {
```

File: compiler/test/suites/numbers.re [hunk 7]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+    expect.string(Warnings.message(FromNumberLiteralF32("5.5"))).toMatch(
+      "5.5f",
+    )
+  });
+  test("float64_fromNumber_err1", ({expect}) => {
```
Comment (phated) [inline compiler/test/suites/numbers.re:155]:
```suggestion
  test("float64_fromNumber_warn1", ({expect}) => {
```

File: compiler/test/suites/numbers.re [hunk 8]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+    expect.string(Warnings.message(FromNumberLiteralF64("5"))).toMatch(
+      "5.d",
+    )
+  });
+  test("float64_fromNumber_err2", ({expect}) => {
```
Comment (phated) [inline compiler/test/suites/numbers.re:160]:
```suggestion
  test("float64_fromNumber_warn2", ({expect}) => {
```

File: compiler/test/suites/numbers.re [hunk 9]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+    expect.string(Warnings.message(FromNumberLiteralF64("5.0"))).toMatch(
+      "5.0d",
+    )
+  });
+  test("float64_fromNumber_err3", ({expect}) => {
```
Comment (phated) [inline compiler/test/suites/numbers.re:165]:
```suggestion
  test("float64_fromNumber_warn3", ({expect}) => {
```

**Review by phated — COMMENTED** [review]
Light nitpicks. If you want to accept these quickly, I can approve and add to the queue.

File: compiler/test/suites/numbers.re [hunk 10]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
   );
+  // well-formedness warnings
+  test("float32_fromNumber_warn1", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF32("5"))).toMatch(
+      "5.f",
+    )
+  });
+  test("float32_fromNumber_warn2", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF32("5."))).toMatch(
+      "5.0f",
```
Comment (phated) [inline compiler/test/suites/numbers.re:147]: I messed up on my suggestions. This needs to change too
```suggestion
      "5.f",
```

File: compiler/test/suites/numbers.re [hunk 11]
```diff
@@ -135,4 +136,35 @@ describe("numbers", ({test, testSkip}) => {
...
+    )
+  });
+  test("float64_fromNumber_warn1", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF64("5"))).toMatch(
+      "5.d",
+    )
+  });
+  test("float64_fromNumber_warn2", ({expect}) => {
+    expect.string(Warnings.message(FromNumberLiteralF64("5."))).toMatch(
+      "5.0d",
```
Comment (phated) [inline compiler/test/suites/numbers.re:162]:
```suggestion
      "5.d",
```

**Review by phated — COMMENTED** [review]
I missed two

**Review by phated — APPROVED** [review]
Excellent work! Thanks for learning enough Reason to write this ❤️

**Pull request merged**